### ISSUE TITLE
Bug: RedirectScheme does not respect X-Forwarded-Proto Header

### ISSUE THREAD

**Issue by mkotsalainen** [traefik/traefik#9112]
### Welcome!

- [X] Yes, I've searched similar issues on [GitHub](https://github.com/traefik/traefik/issues) and didn't find any.
- [X] Yes, I've searched similar issues on the [Traefik community forum](https://community.traefik.io) and didn't find any.

### What did you do?

I've installed traefik via helm on an EKS Cluster. AWS LB does SSL termination and adds X-Forwarded-Proto header. I've added  --entryPoints.websecure.forwardedHeaders.insecure to additionalArguments and I see by debugging that the header is passed on to service. I expect traefik RedirectScheme middleware to respect this header and not perform redirection on http requests with X-Forwarded-Proto header.

### What did you see instead?

Clients end up in redirect loop because traefik keeps redirecting the https traffic (which has been terminated by AWS LB).

I see similar questions relating to this for example: https://github.com/traefik/traefik/issues/5830

When looking at previous questions I don't understand if this is a bug or if you don't intend to support  X-Forwarded-Proto header in RedirectScheme middleware. If not you should probably make that clear in the docs.

### What version of Traefik are you using?

traefik:2.7.1 / helm.sh/chart=traefik-10.21.1

### What is your environment & configuration?

```yaml
* AWS Classic LB
* EKS
* traefik
```

Add more configuration information here.

**Comment by mkotsalainen** [comment]
Just as a clarification, I enable the insecure forwarded headers on both entry points:
```
    - --entryPoints.web.forwardedHeaders.insecure
    - --entryPoints.websecure.forwardedHeaders.insecure
```

**Comment by tomMoulard** [comment]
Fixed in #9121
